reporting/mapreduce_logfile.py

```diff
@@ -0,0 +1,782 @@
+        line contains the line that was attempted to be parsed.
+    Postcondition
+        returns 'BAD_PARSE'
+    '''
+    global lines_that_failed_to_parse
+    lines_that_failed_to_parse.append('Failed to parse for {0} in <\n{1}>'
+                                      .format(value, line))
+    return 'BAD_PARSE'
+
+
+def get_log_name():
+    return 'bad_parse_log/{0}.txt'.format(inspect.stack()[1][3])
+
+
+'''
+        Helper Functions
+'''
+
+
+def substring_between(s, start, finish):
+    '''Find string between two substrings'''
+    end_of_start = s.index(start) + len(start)
+    start_of_finish = s.index(finish, end_of_start)
+    return s[end_of_start:start_of_finish]
+
+'''
+        Extract Data from line in logfile
+'''
+
+
+def get_rtcode(line):
+    '''Obtain a return_code from a line of text
+
+    Precondition: line is a ' ' separated list of data.
+                Return code is the first int in the items from 6 to 11.
+    Postcondition: return return_code
+    '''
+    data = line.split()
+    if(data[8].isdigit()):
+        return data[8]
+    else:
+        return fail_to_parse('rtcode', line)
+
+
+def get_bytes(line):
+    '''Obtain a return_code from a line of text
+
+    Precondition: line is a ' ' separated list of data.
+                Bytes downloaded is the second int in the items from 6 to 12.
+    Postcondition: return bytes_downloaded
+    '''
+    data = line.split()
+    if(data[9].isdigit()):
+        return data[9]
+    else:
+        return fail_to_parse('bytes', line)
+
+
+def get_datetime(line):
+    time_local = substring_between(line, '[', '] "')
+    try:
+        return datetime.datetime.strptime(time_local,
+                                          '%d/%b/%Y:%H:%M:%S -0500')
+    except ValueError:
+        return fail_to_parse('datetime', line)
+
+
+def get_date(line):
+    try:
+        return get_datetime(line).date()
+    except ValueError:
+        return fail_to_parse('date', line)
+
+
+def get_user_email(line):
+    request = substring_between(line, '] "', '" ')
+    request = urllib.unquote(request)
+    try:
+        return substring_between(request, 'orders/', '-')
+    except ValueError:
+        return fail_to_parse('user_email', line)
+
+
+def get_scene_id(line):
+    try:
+        response_after_orderid = substring_between(line, 'orders/', '" ')
+        return substring_between(response_after_orderid, '/', '.tar.gz')
+    except ValueError:
+        try:
+            response_after_orderid = substring_between(line, 'orders/', '" ')
+            return substring_between(response_after_orderid, '/', '.cksum')
+        except ValueError:
+            return fail_to_parse('sceneid', line)
+
+
+def get_order_id(line):
+    request = substring_between(line, '] "', '" ')
+    try:
+        return substring_between(request, 'orders/', '/')
+    except ValueError:
+        return fail_to_parse('orderid', line)
+
+'''
+        Filter helper functions
+'''
+
+
+def n_filters(filters, iterable):
```

Comment:
Was just a function to apply multiple filters to an iterator at once. It has since been removed as the filters have been moved into the mappers.